runtime/src/main/java/org/corfudb/runtime/clients/SequencerHandler.java

```diff
@@ -37,33 +40,95 @@ public SequencerClient getClient(long epoch, UUID clusterID) {
      * The handler and handlers which implement this client.
      */
     @Getter
-    public ClientMsgHandler msgHandler = new ClientMsgHandler(this)
-            .generateHandlers(MethodHandles.lookup(), this);
-
-    /**
-     * For old CorfuMsg, use {@link #msgHandler}
-     * The handler and handlers which implement this client.
-     */
-    @Getter
     public ClientResponseHandler responseHandler = new ClientResponseHandler(this)
             .generateHandlers(MethodHandles.lookup(), this)
             .generateErrorHandlers(MethodHandles.lookup(), this);
 
-    @ClientHandler(type = CorfuMsgType.SEQUENCER_METRICS_RESPONSE)
-    private static Object handleMetricsResponse(CorfuPayloadMsg<SequencerMetrics> msg,
-                                                ChannelHandlerContext ctx, IClientRouter r) {
-        return msg.getPayload();
+    /**
+     * Handle a token response from the server.
+     *
+     * @param msg The token response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router.
+     * @return {@link TokenResponseMsg} sent back from server.
+     */
+    @ResponseHandler(type = PayloadCase.TOKEN_RESPONSE)
+    private static Object handleTokenResponse(CorfuMessage.ResponseMsg msg, ChannelHandlerContext ctx,
+                                              IClientRouter r) {
+        TokenResponseMsg responseMsg = msg.getPayload().getTokenResponse();
+
+        return CorfuProtocolSequencer.getTokenResponse(responseMsg);
     }
 
-    @ClientHandler(type = CorfuMsgType.TOKEN_RES)
-    private static Object handleTokenResponse(CorfuPayloadMsg<TokenResponse> msg,
-                                              ChannelHandlerContext ctx, IClientRouter r) {
-        return msg.getPayload();
+    /**
+     * Handle a bootstrap sequencer response from the server.
+     *
+     * @param msg The bootstrap sequencer response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router.
+     * @return True if ACK, false if NACK.
+     */
+    @ResponseHandler(type = PayloadCase.BOOTSTRAP_SEQUENCER_RESPONSE)
+    private static Object handleBootstrapSequencerResponse(ResponseMsg msg, ChannelHandlerContext ctx,
+                                                           IClientRouter r) {
+        BootstrapSequencerResponseMsg responseMsg = msg.getPayload().getBootstrapSequencerResponse();
+        BootstrapSequencerResponseMsg.Type type = responseMsg.getRespType();
+
+        switch (type) {
+            case ACK:
+                return true;
+            case NACK:
+                return false;
+            // TODO INVALID
+            default:
+                throw new UnsupportedOperationException("Response handler not provided");
+        }
```

Comment:
Maybe this can become a boolean so that we can just directly return the result? I think this is what's being done in the layout PR.

Reply:
Done, Thank you.